rebuildability_check.py

```diff
@@ -0,0 +1,64 @@
+#!/usr/bin/python3
+
+import argparse
+import logging
+import sys
+
+import osc.core
+import osc.conf
+
+import osclib.remote_project
+from osclib.remote_project import RemoteProject
+
+class RebuildabilityChecker(object):
+    def __init__(self, project_str):
+        self.logger = logging.getLogger('RebuildibilityChecker')
+        self.project = RemoteProject.find(project_str) # apiurl should be read from osc.conf.config['apiurl'], osclib config class looks like has different goal?
+
+    def result(self):
+        packages = self.project.get_packages(inherited = True) # package should include also project from which it comes. Recursive means include also inherited packages so reading meta and linked projects are needed
+        self.logger.debug("Packages %s" % [pkg.project_name + " -> " + pkg.name for pkg in packages])
+
+        title = "Testing rebuild of whole parent project"
+        description = "Temporary project including expanded copy of parent to verify if everything can be rebuild from scratch"
+        rebuild_project = self.project.create_subproject("Rebuild", title, description) # how to handle if it exists? Clean it and use? What if we do not have permission to create subproject?
+
+        testing_packages = [package.copy(rebuild_project.name, expand=True) for package in packages]
+        while not all([p.builds.is_finished for p in testing_packages]): # builds is object for handling building and is_finished means that all builds are finished ( or disabled )
+            # write some progress about number of pass, failures, what is waiting, etc. using p.builds query ( ensure it is not memoized )
+            print("Working like a crazy monk")
+
+        # TODO: delete rebuild project or keep it for inspection?
```

Comment:
in principle: all succeeded packages are normally uninteresting. The failed ones you might need to show around. Not sure the yast team ever accepted a "it failed, fix it - that's all I have!" :)

But depending on how often you plan to run this, you surely do not want to have a lot of these projects around as they fill up several hundered GB easily. So reusing them can be key.

Reply:
Maybe it can be even parameter specific e.g. for SLE-15-SP4 I expect we want to keep it and just reuse project when we want another rebuild test. For some other usage like maintenance update, different behavior can make sense.

Reply:
Before you discuss such things, learn how long it takes to create and recompile a single project. So before you start to worry about reports and cleanup, just create one of these beasts

Reply:
yep, that is a goal of this tool, to start playing with it and as side-effect create a bit of high level API that helps with whatever we decided to do.